pkg/bytecode/compiler.go

```diff
@@ -241,6 +243,175 @@ func (c *Compiler) compileBlockStatement(block *parser.BlockStatement) error {
 	return nil
 }
 
+func (c *Compiler) compileForStatement(stmt *parser.ForStmt) error {
```

Comment:
I tried quite hard to unify compileIterRange and compilerStepRange, but the two functions are just different enough that any code sharing begins to look really awkward.

Reply:
I managed to unify these largely (with a separate preamble for the two range types) by pushing a lot of the loop comparison logic into the operations.